checks/prometheus_check.py

```diff
@@ -0,0 +1,232 @@
+
+    def poll(self, endpoint, pFormat=PrometheusFormat.PROTOBUF, headers={}):
+        """
+        Polls the metrics from the prometheus metrics endpoint provided.
+        Defaults to the protobuf format, but can use the formats specified by
+        the PrometheusFormat class.
+        Custom headers can be added to the default headers.
+        """
+        if 'accept-encoding' not in headers:
+            headers['accept-encoding'] = 'gzip'
+        if pFormat == PrometheusFormat.PROTOBUF:
+            headers['accept'] = 'application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited'
+
+        req = requests.get(endpoint, headers=headers)
+        req.raise_for_status()
+        return req.content
+
+    def _submit_metric(self, metric_name, message, send_histograms_buckets=True, labels_mapper={}, custom_tags=None):
+        """
+        For each metric in the message, report it as a gauge with all labels as tags
+        except if a labels dict is passed, in which case keys are label names we'll extract
+        and corresponding values are tag names we'll use (eg: {'node': 'node'}).
+
+        Histograms generate a set of values instead of a unique metric.
+        send_histograms_buckets is used to specify if yes or no you want to
+            send the buckets as tagged values when dealing with histograms.
+
+        If the `labels_mapper` dictionnary is provided, the metrics labels names
+        in the `labels_mapper` will use the corresponding value as tag name.
+
+        `custom_tags` is an array of 'tag:value' that will be added to the
+        metric when sending the gauge to Datadog.
+        """
+        if message.type < len(self.METRIC_TYPES):
+            for metric in message.metric:
+                if message.type == 4:
+                    self._submit_gauges_from_histogram(metric_name, metric, send_histograms_buckets, labels_mapper, custom_tags)
+                elif message.type == 2:
+                    self._submit_gauges_from_summary(metric_name, metric, labels_mapper, custom_tags)
+                else:
+                    val = getattr(metric, self.METRIC_TYPES[message.type]).value
+                    self._submit_gauge(metric_name, val, metric, labels_mapper, custom_tags)
+
+        else:
+            self.log.error("Metric type {} unsupported for metric {}.".format(message.type, message.name))
+
+    def _submit_gauge(self, metric_name, val, metric, labels_mapper=None, custom_tags=None):
+        """
+        Submit a metric as a gauge, additional tags provided will be added to
+        the ones from the label provided via the metrics object.
+
+        If the `labels_mapper` dictionnary is provided, the metrics labels names
+        in the `labels_mapper` will use the corresponding value as tag name.
+        """
+        _tags = custom_tags
+        if _tags is None:
+            _tags = []
+        for label in metric.label:
```

Comment:
Basically it would end up like:
``` _tags = custom_tags
        if _tags is None:
            _tags = []
        for label in metric.label:
            if exclude_labels is not None and label.name not in exclude_labels:
                tag_name = label.name
                if labels_mapper is not None and label.name in labels_mapper:
                    tag_name = labels_mapper[label.name]
                _tags.append('{}:{}'.format(tag_name, label.value))
        self.gauge('{}.{}'.format(self.NAMESPACE, metric_name), val, _tags)
```
Makes sense?